### PR TITLE
chore(checkbox): add new ui state

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**

### CODE REVIEW

File: src/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
+      <StyledCheckbox id={`${id}-icon`} checkedState={checkedState}>
+        {checkedState === SelectionState.selected &&
+          (<Icon  viewBox="0 0 24 24">
+            <polyline points="20 6 9 17 4 12" />
+          </Icon>)
+        }
+        {checkedState === SelectionState.partiallySelected &&
+          (<Icon width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <rect x="6" y="12" width="12" height="1" />
+          </Icon>)
```
Comment (belom88) [inline src/components/checkbox/checkbox.tsx:97]: Could you move the `svg` to a separate file? It looks like we used it this way before we started to use `svgr`

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
-const StyledCheckbox = styled.div<{ checked: boolean }>`
+const StyledCheckbox = styled.div<{ checkedState: SelectionState }>`
   width: 24px;
   height: 24px;
-  background: ${({ checked }) =>
-    checked ? color_brand_tertiary : "transparent"};
+  background: ${({ checkedState }) => {
+    switch (checkedState) {
+      case SelectionState.selected:
+      case SelectionState.partiallySelected:
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Yes, Done, Thank you but it is `indeterminate`.

File: src/components/checkbox/checkbox.tsx [hunk 3]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
+    }
+  }};
   border: 1px solid ${color_brand_tertiary};
   border-radius: 4px;
 
   &:hover {
-    background: ${({ checked }) =>
-      checked ? dim_brand_tertinary : "transparent"};
+    background: ${({ checkedState }) => {
+    switch (checkedState) {
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Changed

File: src/components/checkbox/checkbox.tsx [hunk 4]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
 `;
 
-const StyledCheckbox = styled.div<{ checked: boolean }>`
+const StyledCheckbox = styled.div<{ checkedState: SelectionState }>`
   width: 24px;
   height: 24px;
-  background: ${({ checked }) =>
-    checked ? color_brand_tertiary : "transparent"};
+  background: ${({ checkedState }) => {
+    switch (checkedState) {
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Done

File: src/components/checkbox/checkbox.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Done

File: src/components/checkbox/checkbox.tsx [hunk 5]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
+    }
+  }}
     border: 1px solid ${dim_brand_tertinary};
   }
 
   ${Icon} {
-    visibility: ${(props) => (props.checked ? "visible" : "hidden")};
-  }
+    visibility: ${({ checkedState }) => {
+    switch (checkedState) {
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Done

File: src/components/checkbox/checkbox.tsx [hunk 6]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
-  <CheckboxContainer>
-    <HiddenCheckbox id={id} checked={checked} onChange={onChange} {...rest} />
-    <StyledCheckbox checked={checked}>
-      <Icon id={`${id}-icon`} viewBox="0 0 24 24">
-        <polyline points="20 6 9 17 4 12" />
-      </Icon>
-    </StyledCheckbox>
-  </CheckboxContainer>
-);
+export const Checkbox = ({ id, onChange, checked: checkedState, ...rest }: CheckboxProps) => {
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx:51]: Done

File: src/components/checkbox/checkbox.tsx [hunk 7]
```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
...
+  const checked = checkedState == SelectionState.selected;
```
[the remark this answers is not in the document]
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Done

File: src/components/checkbox/checkbox.tsx (hunk 1 above)
Comment (dryabinin-actionengine) [reply src/components/checkbox/checkbox.tsx]: Done

**Review by belom88 — APPROVED** [review]

**Pull request merged**